Settlement - Get Started page (#378)
wormhole-foundation/wormhole-docs: products/settlement/get-started.md

    MAINNET_SOL_PRIVATE_KEY="your_solana_private_key"
    ```

    !!! warning
        If you use a .env file during development, add it to your .gitignore to exclude it from version control. Never commit private keys or mnemonics to your repository.

    Add this to your `.gitignore`:

    ```bash
    node_modules
    .env
    ```

## Configure Wallet Access

The Mayan Swift route requires signing transactions on both EVM and Solana. To handle this cleanly, you’ll write a helper function that:

- Detects the platform (Solana or EVM)
- Loads the correct signer from the environment
- Returns both the signer and a Wormhole-formatted address

Create `src/helpers.ts` and add the following:

```ts title="src/helpers.ts"
--8<-- "code/products/settlement/get-started/snippet-1.ts"
```

- `getEnv()` makes sure required keys are present, or throws a clear error.
- The `getSigner()` function:
    - Checks if the chain is EVM or Solana
    - Loads the appropriate signer using Wormhole SDK helpers
    - Returns both the signer (used to send txs) and address (used as the recipient)

You’ll use this in the next step to load the sender (on Ethereum) and receiver (on Solana).

> ## Perform the Token Swap
> 
> Now you’ll build the script that performs the swap using the Mayan Swift route. Here’s what happens in the `swap.ts` script:
> 
> - **Initialize Wormhole**: Sets up the SDK for Mainnet with EVM and Solana support.
> - **Define chains and tokens**: specifies you're sending native ETH on Base to native SOL on Solana.
> - **Load wallets**: Pulls signers and addresses from your `getSigner()` helper.
> - **Create transfer request**: Tells Wormhole what you're trying to do.
> - **Find routes**: Asks the Mayan Swift resolver to suggest valid ways to perform the transfer.
> - **Transfer parameters** – specifies how much to send and uses default route options.
> - **Validate route**: Checks if your intent is valid (e.g. sufficient liquidity, no errors).
> - **Quote**: Retrieves the expected output and fees.
> - **Initiate**: Sends the transaction on Ethereum (Base).
> - **Complete**: waits for the VAA and finalizes the transfer on Solana.
> 
> Add the following code to `src/swap.ts`:
> 
> ```ts title="src/swap.ts"
> --8<-- "code/products/settlement/get-started/snippet-2.ts"
> ```
> 
> ## Add a Run Script
> 
> To simplify running the swap script, update your `package.json` with the following:
> 
> ```json title="package.json"
> {
>   "scripts": {
>     "swap": "npx tsx src/swap.ts"
>   }
> }
> ```
> 
> ## Run the Script
> 
> Once everything is in place, you can execute the swap script with:
> 
> ```bash
> npm run swap
> ```
> If successful, you’ll see terminal output like this:
> 
> ```bash
> Validated: { valid: true, ... }
> Quote: { success: true, ... }
> Initiated transfer with receipt: ...
> Current Transfer State: DestinationFinalized
> ```
> 
> Congrats!!! You've just completed a cross-chain token swap from Ethereum to Solana using Settlement!
> 

Reviewer: ```suggestion
## Perform a Token Swap

This section shows you how to perform a token swap using the Mayan Swift route. You will define a helper function to configure the source and destination chain signers.

Then, you'll create a script that initiates a transfer on Ethereum, uses the Mayan Swift resolver to find valid routes, sends the transaction, and completes the transfer on Solana.

1. Open `helper.ts` and define the `getSigner` utility function to load private keys, instantiate signers for Ethereum and Solana, and return the signers along with the Wormhole-formatted address:

    ```ts title="src/helpers.ts"
    --8<-- "code/products/settlement/get-started/snippet-1.ts"
    ```

2. In `swap.ts`, add the following script, which will handle all of the logic required to perform the token swap: 

    ```ts title="src/swap.ts"
    --8<-- "code/products/settlement/get-started/snippet-2.ts"
    ```

3. Execute the script to initiate and complete the transfer:

    ```bash
    npx tsx src/swap.ts
    ```

    If successful, you’ll see terminal output like this:

    INSERT_TERMINAL_ELEMENT_WITH_OUTPUT

Congratulations! You've just completed a cross-chain token swap from Ethereum to Solana using Settlement.
```

Make sure to replace  INSERT_TERMINAL_ELEMENT_WITH_OUTPUT with a snippet that uses the terminal element (like Token Bridge) !!

## Customize the Integration

You can tailor the example to your use case by adjusting:

> - **Tokens and chains** – use `getSupportedTokens()` to explore what's available
> - **Transfer settings** – update the amount or route parameters
> - **Signer management** – modify `src/helpers.ts` to integrate with your preferred wallet setup
> - **Source and destination chains**: modify `sendChain` and `destChain` in `swap.ts`
> - **Amount and transfer settings**: adjust amount to suit your needs
> 

Reviewer: ```suggestion
- **Tokens and chains**: Use `getSupportedTokens()` to explore what's available.
- **Source and destination chains**: Modify `sendChain` and `destChain` in `swap.ts`.
- **Transfer settings**: Update the amount or route parameters.
- **Signer management**: Modify `src/helpers.ts` to integrate with your preferred wallet setup.

```

Reviewer: Amount and transfer settings was duplicated

## Next Steps

Once you've chosen a path, follow the corresponding guide to start building:

- [**Integrate with Liquidity Layer**](/docs/products/settlement/guides/liquidity-layer/){target=\_blank} – interact directly with routers for flexible protocol-level control

> - [**Use Mayan Swift with the SDK**](TODO){target=\_blank} – plug into Settlement using the [TypeScript SDK](https://www.npmjs.com/package/@wormhole-foundation/sdk){target=\_blank} for rapid integration

Reviewer: ```suggestion
<!-- - [**Use Mayan Swift with the SDK**](TODO){target=\_blank} – plug into Settlement using the [TypeScript SDK](https://www.npmjs.com/package/@wormhole-foundation/sdk){target=\_blank} for rapid integration -->
```

I don't think we'll have this ready in time.

This page compares two supported integration options, [Liquidity Layer](/docs/products/settlement/concepts/architecture/#wormhole-liquidity-layer){target=\_blank} and [Mayan Swift](/docs/products/settlement/concepts/architecture/#mayan-swift){target=\_blank}, to help you choose the best fit for your use case.

## Choose Your Integration Path

Depending on your use case, you can choose between different settlement routes. Here's a quick comparison:

<div markdown class="full-width">

::spantable::

| Integration Path               | Best for                              | Chains             | Typescript SDK Compatible | Flexibility| 
|--------------------------------|---------------------------------------|--------------------|--------------------|--------------------|
| Liquidity Layer @span | dApps needing deeper control or protocol-level integration| Testnet + Mainnet | ❌ | :green_circle: :green_circle: :green_circle: <br> High | 
| Mayan Swift @span              | dApps that want fast integration with minimal setup | Mainnet           | ✅ | :green_circle: :white_circle: :white_circle: <br> Low | 

> 

Reviewer 2: to add for comparison: 
Mayan swift has a typical intents architecture, good for speed (12s transactions) but costly and less scalable because it needs balancing of assets
Liquidity layer has a hub and spoke model so uses USDC as shuttle which makes it more scalable, a bit more slow (15-25s transactions)

---
title: Get Started
description: Compare integration paths for Wormhole Settlement and choose the best option for your use case—app, router, or protocol-level.
categories: Settlement, Transfer
---

# Get Started

## Introduction

> Wormhole Settlement enables fast, intent-based cross-chain asset transfers across multiple blockchains like Ethereum, Solana, Sui, and more. This page helps you choose the right integration path for your application and prepare your environment for development.

Reviewer: > prepare your environment for development

It currently does not do this, so just a reminder that this is an item to circle back to

> Settlement offers multiple integration routes, each designed for different use cases, levels of complexity, and supported ecosystems.

Reviewer: This feels like it should be before the "This page helps you..." sentence

It's like "Settlement does this... this page does this.. and settlement does this." So let's put all the "settlement does this" stuff together and the "this page does this" last

## Choose Your Integration Path

Depending on your use case, you can choose between several settlement routes. Here's a quick comparison:

<div markdown class="full-width">

::spantable::

| Integration Path               | Best for                              | Chains             | Technology         | Ease of Integration| 
|--------------------------------|---------------------------------------|--------------------|--------------------|--------------------|

> | Liquidity Layer @span | DApps with flexible settlement and fast settlement finality| Testnet + Mainnet | Typescript | :green_circle: :white_circle: :white_circle: <br> Low | 

Reviewer: TypeScript, please update all to capitalize the S

| Mayan Swift @span              | Stablecoin transfers                              | Mainnet           | Typescript | :green_circle: :white_circle: :white_circle: <br> Low | 

> | MCTP @span        | Programmable cross-chain USDC payments using Circle CCTP | To be defined | Typescript | :green_circle: :green_circle: :white_circle: <br> Moderate | 

Reviewer: What do you mean by to be defined?

Reviewer: What makes this moderate?

| Solver Infra @span             | Liquidity providers fulfilling intents via Solana Matching Engine | Solana Devnet | Typescript, Shell | :green_circle: :green_circle: :white_circle: <br> Moderate| 

::end-spantable::

</div>

## Next Steps

> Once you choose your integration path, you can follow the specific guide that matches your needs:

Reviewer 3: For the call to actions, would it be possible to use more evoking verbs/verbal phrases. I think integrate was a great choice for the first one.   I've added some possibilities below: 
Set up - initiate, establish
Learn - discover, explore
See - refer to

 - [Build on the Wormhole Liquidity Layer](){target=\_blank} – integrate with Wormhole’s chain abstraction layer for fast cross-chain applications
 - [Run a Wormhole Settlement Solver](){target=\_blank} – set up and operate a Solver node to participate in fulfilling cross-chain intents
 - [Understand the Settlement Architecture](){target=\_blank} – learn about the different swap protocols (Liquidity Layer, Mayan Swift, MCTP)

>  - [Review Settlement FAQs](){target=\_blank} – see common questions about auctions, fallback behavior, and settlement guarantees

Reviewer: For any links you're unsure of, please add TODO so we can circle back to all of these items before launch
